zulipterminal/model.py

```diff
@@ -945,24 +947,44 @@ def get_stream_by_id(streams: List[StreamData], stream_id: int
                         for user_id in user_ids:
                             subscribers.remove(user_id)
 
+    @asynch
+    def _show_typing_animation(self, recipient: Dict[str, Any]) -> None:
+        start_time = datetime.now()
+        # Restore the footer if there hasn't been a corresponding typing
+        # 'stop' event after a 'start' event for greater than 15 seconds.
+        max_duration = timedelta(seconds=TYPING_STARTED_EXPIRY_PERIOD)
+
+        for dots in itertools.cycle(['', '.', '..', '...']):
+            if not self.is_recipient_typing:
+                break
+            elif datetime.now() - start_time > max_duration:
```

Comment:
This means that even if the recipient is still typing post 15 seconds, the user won't get any feedback, right?

Reply:
This means if the recipient hasn't sent a `stop` (which the senders client regularly sends if the user is idle or escaped out of compose box) for some reason like client crash etc, ZT keeps showing `typing...` .
If the recipient still is typing its the clients responsibility to tell ZT `start` and it will. So, ZT re-displays `typing...`

Reply:
Though I see now that I haven't reset `is_recipient_typing` to `False`. I think this doesn't change the logic but should I reset anyway?

Reply:
My concern is that we do not check whether we have a new `op=start` and just break regardlessly post `TYPING_STARTED_EXPIRY_PERIOD`. What would happen if we have another `op=start` event post `TYPING_STARTED_WAIT_PERIOD`?

Reply:
oooh! I haven't considered the case where a `start` is received while the async function is running resulting in 2 of them running parallely. Maybe a boolean to fix this?
And when a new `start` is received after 15 seconds this process can restart. So I don't see an issue while breaking after 15 sec, did I miss something?

Reply:
I have used the boolean `is_typing_animation_in_progress` to handle the case where 2 of them run in parallel.
It is a bit long but I didn't want to put `animation_in_progress` or `is_animating` as it doesn't tell us about what is animating.
Even `typing_in_progress` can seem similar to `is_recipient_typing`?

Reply:
@preetmishra If you could look at this particular discussion again I would mark it as resolved.